client/volume_future.go

```diff
@@ -0,0 +1,103 @@
+package client
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"net/http"
+	"time"
+
+	"code.cloudfoundry.org/lager"
+	"github.com/tedsuo/rata"
+
+	"github.com/concourse/baggageclaim"
+	"github.com/concourse/baggageclaim/volume"
+)
+
+type volumeFuture struct {
+	client *client
+	handle string
+	logger lager.Logger
+}
+
+func (f *volumeFuture) Wait() (baggageclaim.Volume, error) {
+	request, err := f.client.requestGenerator.CreateRequest(baggageclaim.CreateVolumeAsyncCheck, rata.Params{
+		"handle": f.handle,
+	}, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	backoffFactor := uint(0)
+
+	for {
+		response, err := f.client.httpClient(f.logger).Do(request)
+		if err != nil {
+			return nil, err
+		}
+
+		if response.StatusCode == http.StatusNoContent {
+			response.Body.Close()
+
+			if backoffFactor != 0 {
+				time.Sleep(time.Duration(rand.Intn((1<<backoffFactor)-1)) * time.Second)
```

Comment:
Seconds may be a bit too long. How about a baseline of 10ms, just to make something up? I figure at least in the common case of creating an empty volume it should complete really quickly.

I'd recommend taking a look at https://github.com/cenkalti/backoff - we use it elsewhere and is easier to use at a high level. See https://godoc.org/github.com/cenkalti/backoff#ExponentialBackOff